crates/parser/src/grammar/types.rs

```diff
@@ -283,17 +283,21 @@ pub(super) fn path_type(p: &mut Parser) {
 // type B = crate::foo!();
 fn path_or_macro_type_(p: &mut Parser, allow_bounds: bool) {
     assert!(paths::is_path_start(p));
+    let r = p.start();
     let m = p.start();
+
     paths::type_path(p);
 
     let kind = if p.at(T![!]) && !p.at(T![!=]) {
         items::macro_call_after_excl(p);
-        MACRO_CALL
+        m.complete(p, MACRO_CALL);
```

Comment:
this constructs:

```
MACRO_TYPE
    MACRO_CALL
        ...
```

for macros in type positions. i hope this is correct, since the introduction of MacroType did not update this part.